grails-app/controllers/org/pih/warehouse/product/ProductController.groovy

```diff
@@ -393,7 +393,7 @@ class ProductController {
             }
         }
 
-        if (!productInstance.hasErrors() && productInstance.save(flush: true)) {
+        if (!productInstance.hasErrors() && packageInstance.validate() && productInstance.save(flush: true)) {
```

Comment:
This seems unnecessary or at least overly cautious. We should be able to rely on just the save to do what we want here. Or find a common pattern used by other Grails 3 projects.